### ISSUE TITLE
ComboboxControl: the Reset button doesn't work when using the keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63408]
### Description

Discovered while testing the 'Parent' setting for a post of type page.

![Screenshot 2024-07-11 at 10 06 45](https://github.com/WordPress/gutenberg/assets/1682452/10431d50-6841-4add-9519-d4acb4fc37da)

Can be reproduced on the [storybook](https://wordpress.github.io/gutenberg/?path=/story/components-comboboxcontrol--default&args=allowReset:!true) as well, by enabling the `allowReset` prop.

When using the keyboard, pressing the `Enter` key on the optional Reset button doesn not reset the combobox. Pressing the Spacebar key does work.

Turns out when pressing the `Enter` key, the `keydown` event attached on the container div element fires and actually _sets_ the currently selected suggestion.

### Step-by-step reproduction instructions

- Edit a page and use the `Parent` setting with the keyboard.
- Alternatively, test on the storybook, make sure the `allowReset` prop is set to `true`: https://wordpress.github.io/gutenberg/?path=/story/components-comboboxcontrol--default&args=allowReset:!true
- Select a suggestion.
- Click the reset button with your mouse.
- The combobox is reset.
- I would expect the same behavior when using the keyboard.
- Select a suggestion.
- Move focus to the Reset button.
- Press the Spacebar key and observe it does reset the combobox.
- Repeat, but this time press the Enter key and observe it does _not_ reset the comobox. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
